mitiq/zne/inference.py

```diff
@@ -137,7 +130,17 @@ def mitiq_polyfit(
 
     Raises:
         ExtrapolationWarning: If the extrapolation fit is ill-conditioned.
+        ExtrapolationError: if the deg > len(scale_factors) - 1.
     """
+
+    # This check may be overkilling.
+    if deg > len(scale_factors) - 1:
+        raise ExtrapolationError(
+            "Extrapolation order is too high. The order cannot exceed "
+            f"len(scale_factors) - 1 but order = {deg} and "
+            f"len(scale_factors) = {len(scale_factors)}."
+        )
```

Comment:
I wouldn't include this in `mitiq_polyfit`. If anything, I would raise a warning, but top suggestion is to allow any degree.